### PR TITLE
DEPR: Hide deprecated attrs _AXIS_NAMES & _AXIS_NUMBERS

### LINKED ISSUE #38722: FutureWarning: _AXIS_NAMES has been deprecated.
At random intervals I keep getting the following error.

> \.vscode\extensions\ms-python.python-2020.12.424452561\pythonFiles\lib\python\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_resolver.py:193: FutureWarning: _AXIS_NUMBERS has been deprecated.
>   attr = getattr(var, name)
> \.vscode\extensions\ms-python.python-2020.12.424452561\pythonFiles\lib\python\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_resolver.py:193: FutureWarning: _AXIS_NAMES has been deprecated.
>   attr = getattr(var, name)

This seems to be refering to how I am using a dataframe in pandas. Though, due to the delays, I am not sure what specifically. I have found only one other refernce to this issue on the internet [https://intellij-support.jetbrains.com/hc/en-us/community/posts/360009512560-PyDev-FutureWarning-Django-console](url)
Saying they believe it occurs on `df.columns`
I'm running my project though a conda environment with:           

>        Version                   Build  Channel
> absl-py                   0.9.0                    pypi_0    pypi
> astor                     0.8.1                    pypi_0    pypi
> astunparse                1.6.3                    pypi_0    pypi
> beautifulsoup4            4.9.1                    py37_0
> brotlipy                  0.7.0           py37he774522_1000
> bzip2                     1.0.8                he774522_0
> ca-certificates           2020.7.22                     0
> cachetools                4.1.1                    pypi_0    pypi
> certifi                   2020.6.20                py37_0
> cffi                      1.14.3           py37h7a1dbc1_0
> chardet                   3.0.4                 py37_1003
> conda                     4.8.5                    py37_0
> conda-build               3.18.11                  py37_0
> conda-package-handling    1.6.1            py37h62dcd97_0
> console_shortcut          0.1.1                         4
> cryptography              3.1.1            py37h7a1dbc1_0
> filelock                  3.0.12                     py_0
> gast                      0.2.2                    pypi_0    pypi
> glob2                     0.7                        py_0
> google-auth               1.18.0                   pypi_0    pypi
> google-auth-oauthlib      0.4.1                    pypi_0    pypi
> google-pasta              0.2.0                    pypi_0    pypi
> grpcio                    1.30.0                   pypi_0    pypi
> h5py                      2.10.0                   pypi_0    pypi
> idna                      2.10                       py_0
> importlib-metadata        1.7.0                    pypi_0    pypi
> jinja2                    2.11.2                     py_0
> keras-applications        1.0.8                    pypi_0    pypi
> keras-preprocessing       1.1.2                    pypi_0    pypi
> libarchive                3.4.2                h5e25573_0
> libiconv                  1.15                 h1df5818_7
> liblief                   0.10.1               ha925a31_0
> libxml2                   2.9.10               h464c3ec_1
> lz4-c                     1.9.2                h62dcd97_1
> markdown                  3.2.2                    pypi_0    pypi
> markupsafe                1.1.1            py37hfa6e2cd_1
> menuinst                  1.4.16           py37he774522_0
> numpy                     1.19.0                   pypi_0    pypi
> oauthlib                  3.1.0                    pypi_0    pypi
> openssl                   1.1.1h               he774522_0
> opt-einsum                3.2.1                    pypi_0    pypi
> pip                       20.2.3                   py37_0
> pkginfo                   1.5.0.1                  py37_0
> powershell_shortcut       0.0.1                         3
> protobuf                  3.12.2                   pypi_0    pypi
> psutil                    5.7.2            py37he774522_0
> py-lief                   0.10.1           py37ha925a31_0
> pyasn1                    0.4.8                    pypi_0    pypi
> pyasn1-modules            0.2.8                    pypi_0    pypi
> pycosat                   0.6.3            py37he774522_0
> pycparser                 2.20                       py_2
> pyopenssl                 19.1.0                     py_1
> pysocks                   1.7.1                    py37_1
> python                    3.7.7                h81c818b_4
> python-libarchive-c       2.9                        py_0
> pytz                      2020.1                     py_0
> pywin32                   227              py37he774522_1
> pyyaml                    5.3.1            py37he774522_1
> requests                  2.24.0                     py_0
> requests-oauthlib         1.3.0                    pypi_0    pypi
> rsa                       4.6                      pypi_0    pypi
> ruamel_yaml               0.15.87          py37he774522_1
> scipy                     1.4.1                    pypi_0    pypi
> setuptools                49.6.0                   py37_1
> six                       1.15.0                     py_0
> soupsieve                 2.0.1                      py_0
> sqlite                    3.33.0               h2a8f88b_0
> style                     1.1.0                    pypi_0    pypi
> tensorboard               1.15.0                   pypi_0    pypi
> tensorboard-plugin-wit    1.7.0                    pypi_0    pypi
> tensorflow                2.2.0                    pypi_0    pypi
> tensorflow-estimator      1.15.1                   pypi_0    pypi
> tensorflow-gpu            1.15.0                   pypi_0    pypi
> termcolor                 1.1.0                    pypi_0    pypi
> tqdm                      4.49.0                     py_0
> update                    0.0.1                    pypi_0    pypi
> urllib3                   1.25.10                    py_0
> vc                        14.1                 h0510ff6_4
> vs2015_runtime            14.16.27012          hf0eaf9b_3
> werkzeug                  1.0.1                    pypi_0    pypi
> wheel                     0.35.1                     py_0
> win_inet_pton             1.1.0                    py37_0
> wincertstore              0.2                      py37_0
> wrapt                     1.12.1                   pypi_0    pypi
> xz                        5.2.5                h62dcd97_0
> yaml                      0.2.5                he774522_0
> zipp                      3.1.0                    pypi_0    pypi
> zlib                      1.2.11               h62dcd97_4
> zstd                      1.4.5                h04227a9_0

Anyone got any ideas?

### PR DESCRIPTION
**Pull request by topper-123**
- [x] closes #38722
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Users that do e.g. `[getattr(pd.DataFrame, x) for x in dir(pd.DataFrame())]` or `inspect.getmembers(pd.DataFrame())` currently get an unfriendly deprecation warning. This fixes that by adding `_AXIS_NAMES` & `_AXIS_NUMBERS` to `_hidden_attrs`.

@JSunRae, is it possible for you to add this to your pandas source code and see if this fixes your problem?

xref: #33637

### CODE REVIEW

File: pandas/tests/frame/test_api.py [hunk 1]
```diff
@@ -337,12 +337,10 @@ def test_constructor_expanddim_lookup(self):
...
 
         with warnings.catch_warnings(record=True) as wrn:
-            # _AXIS_NUMBERS, _AXIS_NAMES lookups
             inspect.getmembers(df)
 
-        # some versions give FutureWarning, others DeprecationWarning
-        assert len(wrn)
-        assert any(x.category in [FutureWarning, DeprecationWarning] for x in wrn)
+        # some versions may give FutureWarning, others DeprecationWarning
+        assert not len(wrn)
```
Comment (jorisvandenbossche) [inline pandas/tests/frame/test_api.py:343]: This seems like unrelated to the actual test for constructor_expanddim. Maybe split it in two separate tests?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks!

**Review by jorisvandenbossche — APPROVED** [review]
Thanks for the update!

We could maybe add a small 1.2.1 whatsnew note, not sure if it is worth it

File: pandas/tests/frame/test_api.py [hunk 2]
```diff
@@ -330,19 +330,20 @@ def test_set_flags(self, allows_duplicate_labels, frame_or_series):
...
         df = DataFrame()
 
+        with pytest.raises(NotImplementedError, match="Not supported for DataFrames!"):
+            df._constructor_expanddim(np.arange(27).reshape(3, 3, 3))
+
+    @skip_if_no("jinja2")
+    def test_inspect_getmembers(self):
+        # GH38740
+        df = DataFrame()
+
```
Comment (jreback) [inline pandas/tests/frame/test_api.py:345]: could use assert_produces_warning(None) here, but really nobd.

**Review by jreback — CHANGES_REQUESTED** [review]
i think a note is warrented here in 1.2.1 in the deprecation section is fine

**Comment by topper-123** [comment]
Updated.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @topper-123 very nice!

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x
